build.gradle

```diff
@@ -118,9 +121,34 @@ dependencies {
     testCompile 'org.powermock:powermock-api-mockito2:2.+'
     
     //WA Libraries
-    implementation 'org.wycliffeassociates:8woc2018-common'
-    implementation 'org.wycliffeassociates:kotlin-resource-container'
-    implementation 'com.github.WycliffeAssociates:jdenticon-kotlin:-SNAPSHOT'
+    compile 'org.wycliffeassociates:8woc2018-common'
+    compile 'org.wycliffeassociates:kotlin-resource-container'
+    compile 'com.github.WycliffeAssociates:jdenticon-kotlin:-SNAPSHOT'
+}
+
+//Application plugin setting
+mainClassName = "org.wycliffeassociates.otter.jvm.app.MainKt"
+
+//tell gradle what to put in the jar
+jar {
+    manifest {
+        attributes(
+        'Class-Path': configurations.compile.collect { it.getName() }.join(' '),
+        'Main-Class': mainClassName
+        )
+    }
+    from (configurations.compile.collect { zipTree(it) }) {
+        exclude "META-INF/MANIFEST.MF"
+        exclude "META-INF/*.SF"
+        exclude "META-INF/*.DSA"
+        exclude "META-INF/*.RSA"
+    }
+}
+
+jfx {
```

Comment:
What is this task for? I understand the `jar` one, but not quite sure what the `jfx` one does

Reply:
@mbr4477 I actually just copied that from one of @sarabiaj gradle builds he was using while mine was breaking. I'm not sure if it's necessary because I don't think we're using any of the jfx tasks. I will experiment with removing it. Joe is there a reason I should keep it that you know of?

Reply:
Joe mentioned that this might be important when we encounter systems that don't have JavaFX bundled with the JDK (e.g., ARM)

Reply:
https://github.com/FibreFoX/javafx-gradle-plugin#minimal-setup-of-buildgradle

Reply:
I removed the jfx plugin and it built and ran, but not sure if there's any particular test that should be run before saying we don't need it.

Reply:
the jfx plugin is required for the openjdk and for arm architectures. The only reason it builds and runs without it is because javafx is included in the oracle jdk.

I'm not sure if we are allowed to distribute the oracle jre?

Reply:
I think we can? https://www.oracle.com/technetwork/java/javase/jre-8-readme-2095710.html
> To run your application, a user needs the Java SE Runtime Environment, which is freely available from Oracle. Or, you can redistribute the Java SE Runtime Environment for free with your application, according to the terms of the Oracle Binary Code License Agreement for the Java SE Platform Products.

Reply:
yeah it just may involve including shipping their tos in the installer or something like that.